crates/interpreter/src/valid.rs

```diff
@@ -596,7 +599,7 @@ impl<'a, 'm> Expr<'a, 'm> {
             Else => {
                 match core::mem::replace(&mut self.label().kind, LabelKind::Block) {
                     LabelKind::If(source) => {
-                        self.side_table.stitch(source, self.branch_target(source.result))?
+                        self.side_table.stitch(source, self.branch_target(source.result, true))?
```

Comment:
Similar to the other PR. We can `let mut target = self.branch_target(source.result)` first, increment `target.side_table`, then stitch.